src/GraphQL/Execution/IExecutionStrategy.cs

```diff
@@ -17,8 +17,8 @@ public interface IExecutionStrategy
         Task<ExecutionResult> ExecuteAsync(ExecutionContext context);
 
         /// <summary>
-        /// This methods allows you to control the set of fields that the strategy will execute.
+        /// Returns the children fields for a specified node.
         /// </summary>
-        bool ShouldIncludeNode(ExecutionContext context, IHaveDirectives directives);
+        Fields GetSubFields(ExecutionContext executionContext, ExecutionNode executionNode);
```

Comment:
This seemed better to me. It simplifies `ReadonlyResolveFieldContext` and only exposes a method to retrieve the subfields, rather than a method with implementation details (requiring all of `CollectFrom` to be `static`, and hence unable to be changed my a derived execution strategy).